### PR TITLE
[RFR] First version of api code

### PR DESCRIPTION
**Pull request by floo51**

### CODE REVIEW

File: api/.npmignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# package directories
```
Comment (Kmaschta) [inline api/.npmignore]: Remove this file

File: api/config/development.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+module.exports = {
```
Comment (Kmaschta) [inline api/config/development.js]: Add this file in `.gitignore`

File: api/package.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
+{
+  "name": "comfygure",
+  "version": "0.1.0",
+  "license": "MIT",
+  "scripts": {
+    "test": "make --silent test"
+  },
+  "dependencies": {
+    "co": "^4.6.0",
```
Comment (Kmaschta) [inline api/package.json]: Please use `~` instead of `^`

File: api/package.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    "co-postgres-queries": "^0.1.0",
```
Comment (Kmaschta) [inline api/package.json]: Please use `~` instead of `^`

File: api/package.json [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    "config": "~1.20.1",
+    "object-hash": "^1.1.5",
```
Comment (Kmaschta) [inline api/package.json]: Please use `~` instead of `^`

And I'm not in favor to use this lib

File: api/package.json [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    "uuid": "^3.0.1"
```
Comment (Kmaschta) [inline api/package.json]: Please use `~` instead of `^`

File: api/schema.sql [hunk 7]
```diff
@@ -0,0 +1,70 @@
+CREATE EXTENSION pgcrypto;
```
Comment (Kmaschta) [inline api/schema.sql]: Move this file in `api/var`

File: api/serverless.yml [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+
+plugins:
+  - serverless-webpack
+  - serverless-offline
+custom:
+  webpack: ./webpack.config.babel.js
+  webpackIncludeModules: true # enable auto-packing of external modules
+  serverless-offline:
+    babelOptions:
+      presets: ["es2015-node4", "es2016"]
```
Comment (Kmaschta) [inline api/serverless.yml]: Can't you use the `.babelrc` file here?

File: api/src/domain/configurations/add.js [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+import entriesQueries from '../../queries/entries';
+import versionsQueries from '../../queries/versions';
+import tagsQueries from '../../queries/tags';
+import { get as getVersion } from './version';
+
+export default function* (projectId, environmentName, configName, tagName = 'next') {
+    const { version: lastVersion, tag: lastTag } = yield getVersion(projectId, environmentName, configName, tagName);
+
+    const versionHash = hash({
```
Comment (Kmaschta) [inline api/src/domain/configurations/add.js]: >_>

File: api/src/domain/configurations/add.js [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+        previous: lastVersion.hash,
+        entries: this.event.body,
+    });
+
+    const version = yield versionsQueries.insertOne({
+        hash: versionHash,
+        previous: lastVersion.hash,
+        date: new Date(),
```
Comment (Kmaschta) [inline api/src/domain/configurations/add.js]: `created_date`

File: api/src/domain/configurations/add.js [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+    });
+
+    yield tagsQueries.updateOne(lastTag.id, {
+        version_id: version.id,
+    });
+
+    for (const key in this.event.body) {
```
Comment (Kmaschta) [inline api/src/domain/configurations/add.js]: Use `const key of Object.keys(this.event.body)`

File: api/src/domain/configurations/get.js [hunk 12]
```diff
@@ -0,0 +1,32 @@
+import entriesQueries from '../../queries/entries';
+import { getDefault as getDefaultVersion, get as getVersion } from './version';
+
+
+export default function* (projectId, environmentName, configName, tagName) {
+    const defaultVersion = yield getDefaultVersion(projectId, configName, tagName);
+    const defaultEntries = (yield entriesQueries.findByVersion(defaultVersion.id))
+        .reduce((acc, item) => ({
```
Comment (Kmaschta) [inline api/src/domain/configurations/get.js]: Please move this reducer in a proper and well named function

File: api/src/domain/configurations/get.js [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+            ...acc,
+            [item.key]: item.value,
+        }), {});
+
+    const { configuration, tag, version } = yield getVersion(projectId, environmentName, configName, tagName);
+    const entries = (yield entriesQueries.findByVersion(version.id))
+        .reduce((acc, item) => ({
```
Comment (Kmaschta) [inline api/src/domain/configurations/get.js]: Please move this reducer in a proper and well named function

File: api/webpack.config.babel.js [hunk 14]
```diff
@@ -0,0 +1,23 @@
+const path = require('path');
+const ConfigPlugin = require('webpack-config-plugin');
+
+module.exports = {
+    target: 'node',
+    entry: './src/handler.js',
+    plugins: [
+        new ConfigPlugin({ dir: path.resolve(__dirname, 'config') }),
+        // new webpack.DefinePlugin({
```
Comment (djhi) [inline api/webpack.config.babel.js]: old code spotted

File: api/src/domain/configurations/add.js (hunk 10 above)
Comment (djhi) [inline api/src/domain/configurations/add.js]: `created_at` ;)

**Pull request merged**